src/Keboola/GoogleSheetsWriter/Sheet.php

```diff
@@ -30,59 +30,44 @@ public function __construct(Client $client, Table $inputTable, Logger $logger)
         $this->logger = $logger;
     }
 
-    /**
-     * @param array $sheet
-     * @throws UserException
-     * @throws ApplicationException
-     * @throws \Keboola\Google\ClientBundle\Exception\RestApiException
-     */
-    public function process(array $sheet) : void
+    public function process(array $sheet): array
     {
         try {
             // Update sheets title and grid properties.
-            // This will adjust the column count to the size of the uploaded table,
-            // so the limit of 2M cells can be used efficiently.
-            $spreadsheet = $this->client->getSpreadsheet($sheet['fileId']);
-            $sheetProperties = $this->findSheetPropertiesById($spreadsheet['sheets'], $sheet['sheetId']);
+            // This will adjust the columns and rows count to the size of the uploaded table
+            $sheetProperties = $this->getSheetProperties($sheet['fileId'], $sheet['sheetId']);
             $columnCount = $this->inputTable->getColumnCount();
-            $rowCount = $this->inputTable->getRowCount();
+            $rowCountSrc = $this->inputTable->getRowCount();
 
             if (empty($sheetProperties)) {
                 throw new UserException(sprintf(
-                    'Sheet "%s (%s)" not found in file "%s (%s)"',
+                    'Sheet "%s" (%s) not found in file "%s" (%s)',
                     $sheet['sheetTitle'],
                     $sheet['sheetId'],
                     $sheet['title'],
                     $sheet['fileId']
                 ));
             }
 
-            if ($columnCount * $rowCount > 2000000) {
+            if ($columnCount * $rowCountSrc > 2000000) {
                 throw new UserException('CSV file exceeds the limit of 2000000 cells');
             }
 
             // update columns
-            $gridProperties = [
-                'columnCount' => $columnCount,
-                'rowCount' => $sheetProperties['properties']['gridProperties']['rowCount'],
-            ];
-
-            $this->updateMetadata($sheet, $gridProperties);
+            $rowCountDst = $sheetProperties['properties']['gridProperties']['rowCount'];
+            $this->updateMetadata($sheet, ['columnCount' => $columnCount, 'rowCount' => $rowCountDst]);
```

Comment:
Proč je tady tohle? To čtu jako "updatnu metadata na počet sloupců, které tam chci nahrát a počet řádků, které tam jsou ($rowCountDst). Nemělo by tam být $rowCountSrc?

Reply:
Citas to spravne :) Je to tak schvalne kvoli bugu v Google Sheets. Totiz maximalny pocet buniek v sheete by mal byt 2M. Ale ked chces sheet resiznut na tuto velkost v jednom API calle tak to hodi chybu. Hodi to tusim uz nad 800k buniek. Preto najprv zmenim pocet stlpcov, pocet riadkov ostane rovnaky ($rowCountDst) a potom sa vola druhy resize, ktory upravi pocet riadkov:
https://github.com/keboola/google-sheets-writer/pull/34/files#diff-6c46cb795acaa9222e1dfe92496218a8R66